doc/generator.md

```diff
@@ -0,0 +1,15 @@
+# Generator
+
+## What is generator
+
+**[generator-napi-module](https://www.npmjs.com/package/generator-napi-module)** is a module to quickly generate a skeleton module using
+**N-API**, the new API for Native addons. This module automatically sets up your
+**gyp file** to use **node-addon-api**, the C++ wrappers for N-API and generates 
+a wrapper JS module. Optionally, it can even configure the generated project to 
+use **TypeScript** instead.
```

Comment:
@sampsongao is this correct.  I was not aware it has anything special for TypeScript

Reply:
@mhdawson I just tried to use generator and at one point it ask if I want generate TypeScript wrapper. I attached the image below:
<img width="580" alt="schermata 2017-11-17 alle 11 07 44" src="https://user-images.githubusercontent.com/387506/32942227-b0c0df26-cb88-11e7-87f6-75dcf282edde.png">

Reply:
Ok, I was thinking of the conversion tool not the generator. I stand corrected.